components/Header.tsx

```diff
@@ -0,0 +1,75 @@
+'use client';
+import { mdiDotsHorizontal, mdiMenu } from '@mdi/js';
+import Icon from '@mdi/react';
+import axios, { AxiosError, HttpStatusCode } from 'axios';
+import Link from 'next/link';
+import { useRouter } from 'next/navigation';
+import { useState } from 'react';
+import Modal from './Modal';
+
+export default function Header({ sidebarId }: { sidebarId: string }) {
+  const router = useRouter();
+  const [error, setError] = useState('');
+  const handleLogout = async () => {
+    try {
+      const response = await axios.post('api/auth/logout');
+      console.log(response);
+      if (response.status === HttpStatusCode.Ok) router.replace('/login');
+      else
+        throw new AxiosError(
+          undefined,
+          response.status as unknown as string,
+          response.config,
+          response.request,
+          response
+        );
+    } catch (err) {
+      const axiosErr = err as AxiosError;
+      setError(`${axiosErr.response?.status} - ${axiosErr.response?.statusText}`);
+      document.getElementById('error-modal')?.click();
```

Comment:
There has to be a better way of doing this that doesn't require plain JS. Maybe using a useRef hook, not sure. This type of code looks like something that shouldn't be done.

Reply:
Dale